packages/viem-account-ledger/src/ledger-to-wallet-client.ts

```diff
@@ -0,0 +1,68 @@
+import { zeroRange } from '@celo/base'
+import TransportNodeHid from '@ledgerhq/hw-transport-node-hid'
+import { Chain, createWalletClient, Transport, WalletClientConfig } from 'viem'
+import { DEFAULT_DERIVATION_PATH } from './constants'
+import { ledgerToAccount } from './ledger-to-account'
+import { AddressValidation, LedgerAccount, LedgerWalletClient } from './types'
+import { generateLedger } from './utils'
+
+const ADDRESS_QTY = 1
+
+export async function ledgerToWalletClient<T extends Chain | undefined = undefined>({
+  transport,
+  derivationPathIndexes = zeroRange(ADDRESS_QTY),
+  changeIndexes = [0],
+  baseDerivationPath = DEFAULT_DERIVATION_PATH,
+  ledgerAddressValidation,
+  walletClientOptions,
+}: {
+  transport: TransportNodeHid
+  derivationPathIndexes?: number[]
+  changeIndexes?: number[]
+  baseDerivationPath?: string
+  ledgerAddressValidation?: AddressValidation
+  walletClientOptions: Omit<WalletClientConfig<Transport, T>, 'account'>
+}): Promise<LedgerWalletClient<T>> {
+  const ledger = await generateLedger(transport)
+  const accounts: Promise<LedgerAccount>[] = []
+  validateIndexes(derivationPathIndexes, 'address index')
+  validateIndexes(changeIndexes, 'change index')
+
+  const _baseDerivationPath = baseDerivationPath.startsWith('m/')
+    ? baseDerivationPath.slice(2)
+    : baseDerivationPath
+
+  // https://trezor.io/learn/a/what-is-bip44
+  const [purpose, coinType, accountIndex] = _baseDerivationPath.split('/')
+  for (const changeIndex of changeIndexes) {
+    for (const addressIndex of derivationPathIndexes) {
+      accounts.push(
+        ledgerToAccount({
+          ledger,
+          derivationPathIndex: addressIndex,
+          baseDerivationPath: `${purpose}/${coinType}/${accountIndex}/${changeIndex}`,
+          ledgerAddressValidation,
+        })
+      )
+    }
+  }
+  const walletClient = createWalletClient({
+    ...walletClientOptions,
+    account: await accounts[0],
+  })
+  walletClient.getAddresses = () =>
+    Promise.all(accounts).then((xs) => xs.map((account) => account.address))
```

Comment:
is this the way to define getAddresses on a walletclient? seems like overwriting something that should be immutable. idk. i guess its ok.

Reply:
resolved in slack